src/server.c

```diff
@@ -3763,6 +3783,10 @@ int processCommand(client *c) {
      * condition, to avoid mixing the propagation of scripts with the
      * propagation of DELs due to eviction. */
     if (server.maxmemory && !isInsideYieldingLongCommand()) {
+
+        /* Pause clients on write as long as OOM and pending lazy-free jobs in background */
+        ClientWritePauseDuringOOM();
```

Comment:
My understanding is we want to avoid evictions if we have a background jobs running. If we are over max-memory here, we are still going to evict on `performEvictions()`. I was almost expecting this call to be in `performEvictions` so we could set the pause and bail.

Reply:
Function `ClientWritePauseDuringOOM()` calls to `pauseClients()` which already take care to the low level details of pausing evictions properly. That is, later on the flow, function `performEvictions()` calls at start to `isSafeToPerformEvictions()` which calls to `checkClientPauseTimeoutAndReturnIfPaused()` which returns 0 because earlier `pauseClients()` was called, That is not safe to perform evictions, and flow of function `performEvictions()` ends without doing any evictions.

Reply:
ok, it's a little odd since it **is** safe to perform evictions.
i.e. the **safe** part is about not emitting anything to the replication stream.
but wording aside, i guess the outcome would be the correct one.
i.e. the eviction will get skipped, and this command (if it's a write command) will get paused instead of processed (handled later in processCommand).

Reply:
Maybe I change the function name to `isPerformEvictions()`

Reply:
i rather revert that rename (the name you suggested seems like it has grammar problems).
but let's add a comment here that mentions:
1. if we initiated a pause then isSafeToPerformEvictions will prevent the eviction in the later call
2. if we initiated a pause, and this is a write command, it'll get postponed later in processCommand.